sockeye/data_io.py

```diff
@@ -1426,6 +1570,141 @@ def load_state(self, fname: str):
         self.shard_iter.load_state(fname + ".sharditer")
 
 
+class CurriculumParallelSampleIter(BaseParallelSampleIter):
+    """
+    Goes through the data one shard at a time. The memory consumption is limited by the memory consumption of the
+    largest shard. The order in which shards are traversed is changed with each reset.
+    """
+
+    def __init__(self,
+                 shards_fnames: List[str],
+                 shards_complexity: List[int],
+                 curriculum_update_freq: int,
+                 buckets,
+                 batch_size,
+                 bucket_batch_sizes,
+                 fill_up: str,
+                 source_data_name=C.SOURCE_NAME,
+                 target_data_name=C.TARGET_NAME,
+                 label_name=C.TARGET_LABEL_NAME,
+                 num_factors: int = 1,
+                 dtype='float32') -> None:
+        super().__init__(buckets=buckets, batch_size=batch_size, bucket_batch_sizes=bucket_batch_sizes,
+                         source_data_name=source_data_name, target_data_name=target_data_name,
+                         label_name=label_name, num_factors=num_factors, dtype=dtype)
+        assert len(shards_fnames) > 0
+        self.max_shard_complexity = 0
+        self.update_max_shard_complexity = False
+        self.shards_fnames = list(shards_fnames)
+        self.shards_complexity = list(shards_complexity)
+        self.visible_shards_fnames = []
+        self.shard_index = -1
+        self.updates_processed = 0
+        self.curriculum_update_freq = curriculum_update_freq
+        self.fill_up = fill_up
+
+        self.reset()
+
+    def _load_shard(self):
```

Comment:
does this method differ from ShardedParallelSampleIter? If not why not subclass from that?